registry/paging_test.go

```diff
@@ -0,0 +1,142 @@
+package registry
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+func TestPages(t *testing.T) {
+	http.HandleFunc("/v2/", pingHandler)
+	http.HandleFunc("/v2/example.com/image/tags/list", tagsHandler)
+	http.HandleFunc("/v2/_catalog", catalogHandler)
```

Comment:
Since those handlers aren't repeated anywhere and are used only for tests, you don't need to have them as full variables.

```
http.HandleFunc("/v2/", func(w http.ResponseWriter, r *http.Request) {
  w.Write([]byte("OK"))
}
```